dpctl/__init__.py

```diff
@@ -47,6 +47,7 @@
 __author__ = "Intel Corp."
 
 from ._sycl_core import *
+from ._memory import MemoryUSMShared, MemoryUSMDevice, MemoryUSMHost
```

Comment:
Do we want this or something like `dpctl.memory.MemoryUSMShared` etc.?

Reply:
I am undecided. On one hand, there is an appeal of simplicity of having it in `dpctl`, on another hand, having it there forces import of `numpy`, which currently presents some problems for `dpnp`.

Reply:
my personal preference is to have a memory sub-module under `dpctl`.

Reply:
Ok. We need to close #120 then.

Reply:
What about `cimport` ? Should it also be `cimport dpctl.memory`? If so, perhaps the easiest is to rename `_memory.pxd` and `_memory.pyx` to remove the underscore.